src/pages/SlayerDetails.tsx

```diff
@@ -45,16 +49,32 @@ const SlayerDetailsPage = () => {
     }
   }, [location.state, tokenId]);
 
-  // In the SlayerDetailsPage component
+  // Check if the slayer owner is the staking contract, if so, fetch the original owner
+  useEffect(() => {
+    if (slayer) {
+      if (slayer.owner === WOV_STAKING_ADDRESS) {
+        setLoading(true);
+        setStaked(true);
+        (async () => {
+          const originalOwner = await stakingCheck(slayer.tokenId);
+          setSlayer((prev) => ({
+            ...prev,
+            owner: originalOwner,
+            tokenId: prev?.tokenId || slayer.tokenId,
+          }));
+          setLoading(false);
+        })();
```

Comment:
could lead to an infinite loading state in case of errors, because the loading is never reset if something throws, suggestion (untested):

```js
stakingCheck(slayer.tokenId)
    .then(owner => {
          setSlayer((prev) => ({
            ...prev,
            owner: originalOwner,
            tokenId: prev?.tokenId || slayer.tokenId,
          }));
     })
    .catch(err => console.error(err))
    .finally(() => setLoading(false))

```